domain/src/main/java/com/andersen/repository_criteria_API/user/UserRepositoryImpl.java

```diff
@@ -0,0 +1,100 @@
+package com.andersen.repository_criteria_API.user;
+
+import com.andersen.entity.role.User;
+import com.andersen.exception.DataAccessException;
+import com.andersen.exception.ErrorCode;
+import com.andersen.logger.ConsoleLogger;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.PersistenceContext;
+import jakarta.persistence.TypedQuery;
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.Root;
+import org.springframework.stereotype.Repository;
+
+import java.util.Optional;
+
+@Repository
+public class UserRepositoryImpl implements UserRepository {
+
+    @PersistenceContext
+    private EntityManager entityManager;
+
+    @Override
+    public User createUser(User user) throws DataAccessException {
+        try {
+            entityManager.persist(user);
+            ConsoleLogger.log("Created user: " + user); // Log user creation
+            return user;
+        } catch (Exception e) {
+            ConsoleLogger.log("Failed to create user:"+e.getMessage()); // Log the error
+            throw new DataAccessException("Failed to create user:" + e, ErrorCode.US_002);
+        }
+    }
+
+    @Override
+    public Optional<User> getUserById(Long id) throws DataAccessException {
+        try {
+            User user = entityManager.find(User.class, id);
+            ConsoleLogger.log("Retrieved user by id: " + id); // Log user retrieval
+            return Optional.ofNullable(user);
+        } catch (Exception e) {
+            ConsoleLogger.log("Failed to get user by id:"); // Log the error
+            throw new DataAccessException("Failed to get user by id:" + e, ErrorCode.US_007);
+        }
+    }
+
+    @Override
+    public Optional<User> getUserByUsername(String username) throws DataAccessException {
+        try {
+            // Using Criteria API for type-safe query
+            CriteriaBuilder cb = entityManager.getCriteriaBuilder();
+            CriteriaQuery<User> cq = cb.createQuery(User.class);
```

Comment:
it's the only placr you use criteria API. Ok, let's assume ypu wanted  to try it all. But mo meed to call repos package "repository_criteria_api"